File: 263.ugly-number.js
/*
 * @lc app=leetcode id=263 lang=javascript
 *
 * [263] Ugly Number
 */

// @lc code=start
/**
 * @param {number} n
 * @return {boolean}
 */
var isUgly = function(n) {
    if (n < 0) return false;
    if (n === 0) return false;
    if (n === 1) return true;
    if (n % 2 === 0) return isUgly(n / 2);
    if (n % 3 === 0) return isUgly(n / 3);
    if (n % 5 === 0) return isUgly(n / 5);
    return false;
};
// @lc code=end

File: 263.ugly-number.py
#
# @lc app=leetcode id=263 lang=python
#
# [263] Ugly Number
#

# @lc code=start
class Solution(object):
    def isUgly(self, n):
        """
        :type n: int
        :rtype: bool
        """

        if n == 0:
            return False

        while n % 2 == 0:
            n /= 2

        while n % 3 == 0:
            n /= 3

        while n % 5 == 0:
            n /= 5

        return n == 1

# @lc code=end
